### PR TITLE
Get rid of MultiIndex conversion in IntervalIndex.is_unique

### PR DESCRIPTION
**Pull request by makbigc**
-  xref #24813 
-  passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

Actually, this modification doesn't improve the performance. But the code is more explanatory.

### CODE REVIEW

**Review by jschendel — CHANGES_REQUESTED** [review]
I can't find any existing benchmarks for this in `asv_bench/benchmarks`, so will need to add benchmarks there and show a performance improvement.

**Comment by jreback** [comment]
can you merge master and report on the asv's for this (and add if needbe)

**Comment by makbigc** [comment]
As jschendel said, this change slowed down the performance. IntervalArray.unique and algo.unique also convert to an array of Interval in the first place.
```

All benchmarks:

       before           after         ratio
     [33f91d8f]       [9c23aac9]
     <master>         <is_unique>
+         249±4ns          378±3ms 1519132.92  index_object.IntervalIndexMethod.time_is_unique
```

**Comment by jreback** [comment]
@makbigc the asv's that you added look good. can you remove the actual perf change.

File: asv_bench/benchmarks/index_object.py [hunk 1]
```diff
@@ -181,4 +181,16 @@ def time_get_loc(self):
         self.ind.get_loc(0)
 
 
+class IntervalIndexMethod(object):
+    # GH 24813
```
Comment (jreback) [inline asv_bench/benchmarks/index_object.py]: is this where we asv on is_unique?

**Comment by makbigc** [comment]
I wrote a new `IntervalIndex.is_unique` which did improve the performance.

```
All benchmarks:

       before           after         ratio
     [33f91d8f]       [b382b173]
     <v0.24.0~79>       <is_unique>
-         249±4ns        113±0.5ns     0.45  index_object.IntervalIndexMethod.time_is_unique
```

File: asv_bench/benchmarks/index_object.py [hunk 2]
```diff
@@ -181,4 +181,16 @@ def time_get_loc(self):
...
+    def setup(self):
+        N = 10**5
+        left = np.append(np.arange(N), np.array(0))
+        right = np.append(np.arange(1, N + 1), np.array(1))
+        self.intv = IntervalIndex.from_arrays(left, right)
+
+    def time_is_unique(self):
```
Comment (jreback) [inline asv_bench/benchmarks/index_object.py]: can you add another benchmark with N**3, e.g. s small case

File: pandas/core/indexes/interval.py [hunk 3]
```diff
@@ -463,7 +463,13 @@ def is_unique(self):
         """
         Return True if the IntervalIndex contains unique elements, else False
         """
-        return self._multiindex.is_unique
+        left = self.values.left
```
Comment (jreback) [inline pandas/core/indexes/interval.py]: why isnt the answer just: ``self.left.is_unique & self.right.is_unique``

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/indexes/interval.py (hunk 3 above)
Comment (jschendel) [reply pandas/core/indexes/interval.py]: That's a little too strict; you don't necessarily left or right uniqueness, only pairwise uniqueness, as you can have duplicate endpoints on one side as long as the other side also isn't the same, e.g. `[Interval(0, 1), Interval(0, 2), Interval(0, 3)]` should be unique.

Comment (jreback) [reply pandas/core/indexes/interval.py]: i think u can simply construct a numpy array and check that then via np.unique by first reshaping

**Comment by jreback** [comment]
can you merge master and update

File: pandas/core/indexes/interval.py (hunk 3 above)
Comment (makbigc) [reply pandas/core/indexes/interval.py]: I have tried something below
```
arr = np.array(list(zip(self.values.left, self.values.right)))
np.unique(arr, axis=0)
```
But np.unique cannot filter out the duplicate np.nan

```
In [2]: arr = np.array([1, 3, 3, np.nan, np.nan])                               

In [3]: np.unique(arr)                                                          
Out[3]: array([ 1.,  3., nan, nan])
```

Comment (jreback) [reply pandas/core/indexes/interval.py]: use pd.unique

**Comment by jreback** [comment]
can you merge master

**Comment by makbigc** [comment]
I broke something when merging. This PR is continued in #26391

**Pull request closed without merging**